### PR TITLE
LB-755: Solve 500 in feedback api

### PR DESCRIPTION
**Pull request by shivam-kapila**
# Problem
The `/feedback/user/<user_name>/get-feedback-for-recordings` returns 500 in prod if `recordings` arg is not pased

# Solution
Raise 400 in such cases.

### CODE REVIEW

File: listenbrainz/webserver/views/feedback_api.py [hunk 1]
```diff
@@ -180,9 +180,12 @@ def get_feedback_for_recordings_for_user(user_name):
 
     recordings = request.args.get('recordings')
 
+    if not recordings:
```
Comment (vansika) [inline listenbrainz/webserver/views/feedback_api.py:183]: You should also check if passed recording msids are valid uuids.

**Review by vansika — CHANGES_REQUESTED** [review]

File: listenbrainz/webserver/views/feedback_api.py (hunk 1 above)
Comment (shivam-kapila) [reply listenbrainz/webserver/views/feedback_api.py:183]: Thats actually checked in pydantic model right now. e.g. https://api.listenbrainz.org/1/feedback/user/shivam-kapila/get-feedback-for-recordings?recordings=foo

Comment (shivam-kapila) [reply listenbrainz/webserver/views/feedback_api.py:183]: Please let me know if this needs change

Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: Ah cool!

Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: I will have a second pass tomorrow :)

Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: https://api.listenbrainz.org/1/feedback/user/Vansika%20Pareek/get-feedback-for-recordings?recording=foo returns 500?

Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: I am not really sure about the SQL query you are using here, but I think the model will check DB/SQL returned `results` and the not the values/args passed to the SQL query which are directly taken from the URL and may not be in the desired format. have a look :)

Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: I wrote `recording` in place of `recordings`. Nevamind. Looks good.

**Review by vansika — APPROVED** [review]
The changes look fine but ig we should wait till tomorrow, you might want/need to add/delete stuff after the scheduled meeting on feedback.

File: listenbrainz/tests/integration/test_feedback_api.py [hunk 2]
```diff
@@ -666,12 +666,17 @@ def test_get_feedback_for_recordings_for_user_invalid_user(self):
         self.assertEqual(response.json["error"], "Cannot find user: nouser")
 
     def test_get_feedback_for_recordings_for_user_no_recordings(self):
-        """ Test to make sure that the API sends 400 if params recordings has nothing. """
+        """ Test to make sure that the API sends 400 if params recordings is not paased or has nothing. """
```
Comment (vansika) [inline listenbrainz/tests/integration/test_feedback_api.py:669]: Passed?

Comment (vansika) [reply listenbrainz/tests/integration/test_feedback_api.py:669]: ".. If param `recordings` is not passed or is empty"

File: listenbrainz/webserver/views/feedback_api.py (hunk 1 above)
Comment (shivam-kapila) [reply listenbrainz/webserver/views/feedback_api.py:183]: This 500 will vanish after the PR is moved to prod. But the other one is still 400.

File: listenbrainz/tests/integration/test_feedback_api.py (hunk 2 above)
Comment (shivam-kapila) [reply listenbrainz/tests/integration/test_feedback_api.py:669]: Makes sense

File: listenbrainz/webserver/views/feedback_api.py (hunk 1 above)
Comment (vansika) [reply listenbrainz/webserver/views/feedback_api.py:183]: Yeah, my bad. I missed the `s` in the spelling :p

**Comment by shivam-kapila** [comment]
@mayhem since the error is still in prod, I think we might want to deploy this fix. Please take a look.

**Comment by mayhem** [comment]
Given that this 500 only occurs when the user uses the API wrong, it doesn't really warrant a separate release.

**Pull request merged**